#–ò—Å–ø–æ–ª—å–∑—É–µ—Ç—Å—è –¥–ª—è —Å—Ç–∞—Ä—Ç–∞ –¥–∏–∞–ª–æ–≥–∞
extends Node

#–ò—Å–ø–æ–ª—å–∑—É–µ—Ç—Å—è –¥–ª—è —Ç–∞–π–º–µ—Ä–∞ (–∫—ç—à)
var time_passed: float = 0.0

#—á–µ—Ä–µ–∑ —Å–∫–æ–ª—å–∫–æ –≤—Ä–µ–º–µ–Ω–∏ –±—É–¥–µ—Ç –≤—ã–∑—ã–≤–∞—Ç—å—Å—è
var time_off: float = 5.0

#–ü—Ä–∏ –∑–∞–ø—É—Å–∫–µ —Å—Ü–µ–Ω—ã, –æ–Ω –∑–∞–ø—É—Å–∫–∞–µ—Ç –∫–æ–º–∞–Ω–¥—ã –∫–æ—Ç–æ—Ä—ã–µ –µ—Å—Ç—å –≤ –Ω–µ–º –≤–æ –≤—Ä–µ–º—è —Å—Ç–∞—Ä—Ç–∞
func _ready() -> void:
	#–ü–æ–¥–ø–∏—Å–∞—Ç—å—Å—è –∏ –ø–æ–ª—É—á–∞—Ç—å —Å–∏–≥–Ω–∞–ª—ã –æ—Ç –Ω–µ–≥–æ
	Dialogic.signal_event.connect(_test_signal)
	
	#–ó–∞–ø—É—Å—Ç–∏—Ç—å —Ç–∞–π–º–ª–∞–π–Ω
	Dialogic.start("timeline1")
	
	#–û—á–∏—Å—Ç–∫–∞ —Å–æ—Ö—Ä–∞–Ω–µ–Ω–∏—è –ø–µ—Ä–µ–¥ –∑–∞–ø—É—Å–∫–æ–º
	Dialogic.Save.delete_slot("autosave")

#–í—ã–∑—ã–≤–∞–µ—Ç—Å—è –∫–∞–∂–¥—ã–π –∫–∞–¥—Ä
func  _process(delta: float) -> void:
	time_passed += delta
	if time_passed >= time_off:
		time_passed = 0.0  # –°–±—Ä–æ—Å —Ç–∞–π–º–µ—Ä–∞
		# –ü–æ–∫–∞–∑—ã–≤–∞–µ—Ç —á—Ç–æ –Ω–∞—Ö–æ–¥–∏—Ç—Å—è –≤ –ø–µ—Ä–µ–º–µ–Ω–Ω–æ–π —Ç–∏–ø–∞ int
		print("–í–Ω—É—Ç—Ä–∏ –∑–Ω–∞—á–µ–Ω–∏–µ —Ö—Ä–∞–Ω–∏—Ç—å—Å—è: (", Dialogic.VAR.Global_variables._test_dia, ")")
		# –ü–æ–∫–∞–∑—ã–≤–∞–µ—Ç –∫–∞–∫–æ–π —Ç–µ–∫—Å—Ç –±—ã–ª –Ω–∞–ø–∏—Å–∞–Ω —á–µ—Ä–µ–∑ –∫–ª–∞–≤–∏–∞—Ç—É—Ä—É
		print("–¢–µ–∫—Å—Ç —Ç–∞–∫–æ–π –Ω–∞—Ö–æ–¥–∏—Ç—Å—è: (", Dialogic.VAR.Keyboard._test, ")")
		
#–ü–æ–ª—É—á–µ–Ω–∏—è –∫–æ–º–∞–Ω–¥ –æ—Ç —Å–∏–≥–Ω–∞–ª–∞
func _test_signal(_argument: String) -> void:
	# –ü—Ä–æ–≤–µ—Ä—è–µ—Ç—Å—è —á—Ç–æ –≤ —Å–∏–≥–Ω–∞–ª–µ –Ω–∞—Ö–æ–¥–∏—Ç—Å—è
	if _argument == "_test_signal":
		print("–í—ã–∑—ã–≤–∞–ª–∏?")
	# –ü—Ä–æ–≤–µ—Ä—è–µ—Ç—Å—è —á—Ç–æ –≤ —Å–∏–≥–Ω–∞–ª–µ –Ω–∞—Ö–æ–¥–∏—Ç—Å—è
	elif _argument == "_not_set":
		print("–¢—ã –∑–∞ –∫–æ–≥–æ –º–µ–Ω—è –¥–µ—Ä–∂–∏—à—å?")

func _on_save_pressed() -> void:
	Dialogic.Save.save("autosave", false, Dialogic.Save.ThumbnailMode.NONE)
	print("‚úÖ –°–æ—Ö—Ä–∞–Ω–µ–Ω–∏–µ –≤—ã–ø–æ–ª–Ω–µ–Ω–æ (autosave)")

func _on_load_pressed() -> void:
	if Dialogic.Save.has_slot("autosave"):
		Dialogic.Save.load("autosave")
		print("üìÇ –ó–∞–≥—Ä—É–∑–∫–∞ –≤—ã–ø–æ–ª–Ω–µ–Ω–∞ (autosave)")
	else:
		print("‚ö†Ô∏è –°–æ—Ö—Ä–∞–Ω–µ–Ω–∏–µ autosave –Ω–µ –Ω–∞–π–¥–µ–Ω–æ!")

func _on_clear_pressed() -> void:
	print("–£–¥–∞–ª–µ–Ω–∏–µ —Å–æ—Ö—Ä–∞–Ω–µ–Ω–∏—è")
	Dialogic.Save.delete_slot("autosave")
